projects/ion/src/lib/card/card.component.spec.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Component,
+  NgModule,
+} from '@angular/core';
+import { fireEvent, screen } from '@testing-library/angular';
+
+let renderFooter = false;
+
+@Component({
+  template: `<ion-button
+    [label]="label"
+    [type]="'primary'"
+    data-testid="buttonBody"
+  ></ion-button>`,
+})
+class ButtonTestComponent {
+  public label = 'Botão no body';
+}
+
+@Component({
+  template: `<h1 data-testid="footerTest">Aqui é o footer</h1>`,
+})
+class FooterTestComponent {}
+
+@Component({
+  template: `<ion-card
+    [configuration]="cardConfi"
+    (events)="cardEvents($event)"
+  ></ion-card>`,
+})
+class CardTestComponent implements AfterViewInit {
+  constructor(private cdr: ChangeDetectorRef) {}
+  ngAfterViewInit(): void {
+    this.cdr.detectChanges();
+  }
+  public cardConfi: IonCard = {
+    body: ButtonTestComponent,
+    header: { title: 'opa' },
+    footer: renderFooter
+      ? { bodyFooter: FooterTestComponent }
+      : { buttons: { primary: { label: 'iea' } } },
+  };
+
+  public cardEvents(event: CardEvent) {
+    this.cardConfi.header.title = `Opa, eu fui clicado evento: ${event.buttonAction}`;
+  }
+}
+
+@NgModule({
+  imports: [CommonModule],
+  declarations: [
+    CardTestComponent,
+    CardIonComponent,
+    ButtonComponent,
+    IonIconComponent,
+    ButtonTestComponent,
+    FooterTestComponent,
+  ],
+  entryComponents: [
+    CardTestComponent,
+    ButtonTestComponent,
+    FooterTestComponent,
+  ],
+})
+class TestModule {}
+
+describe('CardComponent', () => {
+  beforeEach(async () => {
+    TestBed.configureTestingModule({
+      imports: [TestModule],
+    }).compileComponents();
+  });
+
+  it('should render cardComponent', async () => {
+    const fixture = TestBed.createComponent(CardTestComponent);
+    const cardComponent = fixture.componentInstance;
+    fixture.detectChanges();
+    expect(cardComponent).toBeTruthy();
+    fixture.destroy();
+  });
+
+  it('should render body in cardComponent', async () => {
+    const fixture = TestBed.createComponent(CardTestComponent);
+    fixture.detectChanges();
+    expect(screen.getByTestId('buttonBody')).toBeTruthy();
+    fixture.destroy();
+  });
+  it('should change the card header label when the event is triggered', async () => {
+    const fixture = TestBed.createComponent(CardTestComponent);
+    fixture.detectChanges();
+
+    const header = screen.getByTestId('cardHeader');
+    expect(header.textContent).toBe('opa');
+
+    const buttonPrimary = screen.getByTestId('buttonPrimary');
+    fireEvent.click(buttonPrimary);
+    fixture.detectChanges();
+
+    expect(header.textContent).toBe('Opa, eu fui clicado evento: primary');
+    fixture.destroy();
+  });
+
+  it('should render footer in cardComponent', async () => {
+    renderFooter = true;
+    const fixture = TestBed.createComponent(CardTestComponent);
+    fixture.detectChanges();
+    expect(screen.getByTestId('footerTest')).toBeTruthy();
```

Comment:
```suggestion
    expect(screen.getByTestId('footerTest')).toBeTruthy();
```
I believe that toBeInTheDocument make more sense.